### PR TITLE
[KOGITO-5590] Supporting http cloud events on multi addon

### PR DESCRIPTION
**Pull request by fjtirado**
- Removing empty constructor for DefaultEventMarshaller (using injected mapper if no instance is available, which happens for DMN without processes examples) 
- Adding support for Http cloud events on multi addon
- Restoring changes that breaks http cloud events
Please note that CloudEvent modules is already being registered if addon (quarkus or springboot) is enabled.

### CODE REVIEW

File: kogito-codegen-modules/kogito-codegen-api/src/main/java/org/kie/kogito/codegen/api/utils/AddonsConfigDiscovery.java [hunk 1]
```diff
@@ -53,7 +52,7 @@ public static AddonsConfig discover(Predicate<String> classAvailabilityResolver)
         boolean usePrometheusMonitoring = classAvailabilityResolver.test(PROMETHEUS_CLASS);
         boolean useMonitoring = usePrometheusMonitoring || classAvailabilityResolver.test(MONITORING_CORE_CLASS);
         boolean useTracing = classAvailabilityResolver.test(TRACING_CLASS);
-        boolean useCloudEvents = classAvailabilityResolver.test(QUARKUS_CLOUD_EVENTS) || classAvailabilityResolver.test(SPRING_CLOUD_EVENTS);
+        boolean useCloudEvents = classAvailabilityResolver.test(QUARKUS_RESOURCE_CLASS);
```
Comment (ricardozanini) [inline kogito-codegen-modules/kogito-codegen-api/src/main/java/org/kie/kogito/codegen/api/utils/AddonsConfigDiscovery.java:55]: Why removing the SB check?

File: kogito-codegen-modules/kogito-codegen-processes/src/main/resources/class-templates/MessageProducerJavaTemplate.java [hunk 2]
```diff
@@ -27,7 +30,7 @@
 
     Optional<Boolean> useCloudEvents = Optional.of(true);
 
-    EventMarshaller marshaller = new DefaultEventMarshaller();
+    EventMarshaller marshaller = new DefaultEventMarshaller(new ObjectMapper());
```
Comment (ricardozanini) [inline kogito-codegen-modules/kogito-codegen-processes/src/main/resources/class-templates/MessageProducerJavaTemplate.java:33]: How do we get the CloudEvent module registered in this case?

File: quarkus/addons/cloudevents/common/src/main/java/org/kie/kogito/addon/cloudevents/quarkus/AbstractQuarkusCloudEventEmitter.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    private MessageDecorator messageDecorator;
+
+    @PostConstruct
+    private void init() {
+        messageDecorator = MessageDecoratorFactory.newInstance(configBean.useCloudEvents());
+        marshaller = marshallers.isUnsatisfied() ? new DefaultEventMarshaller(mapper) : marshallers.get();
+    }
+
+    protected <T> Message<String> processMessage(T e, Optional<Function<T, Object>> processDecorator) {
+        return this.messageDecorator.decorate(marshaller.marshall(
```
Comment (ricardozanini) [inline quarkus/addons/cloudevents/common/src/main/java/org/kie/kogito/addon/cloudevents/quarkus/AbstractQuarkusCloudEventEmitter.java:56]: I'll need to rebase with this: https://github.com/kiegroup/kogito-runtimes/pull/1476

**Review by ricardozanini — COMMENTED** [review]
Nice! Just left a couple of minor comments.

File: kogito-codegen-modules/kogito-codegen-processes/src/main/resources/class-templates/MessageProducerJavaTemplate.java (hunk 2 above)
Comment (fjtirado) [reply kogito-codegen-modules/kogito-codegen-processes/src/main/resources/class-templates/MessageProducerJavaTemplate.java:33]: To be honest, I do not think this is used. I just changed it so it compiles

File: quarkus/addons/cloudevents/common/src/main/java/org/kie/kogito/addon/cloudevents/quarkus/AbstractQuarkusCloudEventEmitter.java (hunk 3 above)
Comment (fjtirado) [reply quarkus/addons/cloudevents/common/src/main/java/org/kie/kogito/addon/cloudevents/quarkus/AbstractQuarkusCloudEventEmitter.java:56]: this is just a move of shared code between the two addons, not really a change

Comment (fjtirado) [reply quarkus/addons/cloudevents/common/src/main/java/org/kie/kogito/addon/cloudevents/quarkus/AbstractQuarkusCloudEventEmitter.java:56]: but, yes, you will need to replace messageDecorator by messageFactory

File: kogito-codegen-modules/kogito-codegen-api/src/main/java/org/kie/kogito/codegen/api/utils/AddonsConfigDiscovery.java (hunk 1 above)
Comment (fjtirado) [reply kogito-codegen-modules/kogito-codegen-api/src/main/java/org/kie/kogito/codegen/api/utils/AddonsConfigDiscovery.java:55]: There is not CloudEventListener template for Springboot

Comment (ricardozanini) [reply kogito-codegen-modules/kogito-codegen-api/src/main/java/org/kie/kogito/codegen/api/utils/AddonsConfigDiscovery.java:55]: Is this the only reason why we check the `useCloudEvents` then? Interesting...

File: kogito-codegen-modules/kogito-codegen-processes/src/main/resources/class-templates/MessageProducerJavaTemplate.java (hunk 2 above)
Comment (ricardozanini) [reply kogito-codegen-modules/kogito-codegen-processes/src/main/resources/class-templates/MessageProducerJavaTemplate.java:33]: If we have the CDI context, we can try injecting the mapper instead: https://github.com/kiegroup/kogito-runtimes/blob/main/quarkus/addons/cloudevents/common/src/main/java/org/kie/kogito/addon/cloudevents/quarkus/QuarkusKogitoExtensionInitializer.java#L40

**Comment by fjtirado** [comment]
jenkins run examples tests

**Comment by fjtirado** [comment]
jenkins run optaplanner tests

File: kogito-codegen-modules/kogito-codegen-api/src/main/java/org/kie/kogito/codegen/api/utils/AddonsConfigDiscovery.java (hunk 1 above)
Comment (fjtirado) [reply kogito-codegen-modules/kogito-codegen-api/src/main/java/org/kie/kogito/codegen/api/utils/AddonsConfigDiscovery.java:55]: yes, I thought the same ;)

File: kogito-codegen-modules/kogito-codegen-processes/src/main/resources/class-templates/MessageProducerJavaTemplate.java (hunk 2 above)
Comment (fjtirado) [reply kogito-codegen-modules/kogito-codegen-processes/src/main/resources/class-templates/MessageProducerJavaTemplate.java:33]: Thats the problem, there is no CDI context for *JavaTemplate

**Comment by fjtirado** [comment]
jenkins run examples tests

**Comment by fjtirado** [comment]
jenkins run optaplanner tests

**Comment by fjtirado** [comment]
jenkins run examples tests

**Review by ricardozanini — APPROVED** [review]

**Comment by ricardozanini** [comment]
Let's wait for the CI to finish

**Review by tiagodolphine — APPROVED** [review]

**Comment by fjtirado** [comment]
jenkins run examples tests

**Comment by ricardozanini** [comment]
jenkins run examples tests

**Pull request merged**